packages/server/test/helpers.ts

```diff
@@ -91,18 +96,25 @@ after(async function() {
 });
 
 beforeEach(() => {
-  //reset all the state in the global atom, except for the manifest
-  actions.atom.reset(initial => ({ ...initial, manifest }));
-
+  actions.atom.update((state) => ({
+    ...state,
+    agents: initialOrchestratorState['agents'],
+    testRuns: initialOrchestratorState['testRuns'],
+  }));
+
+  afterDestroyPromises = [];
+  
   currentWorld = new World();
 });
 
-afterEach(() => {
+afterEach(async () => {
   if(globalWorld.execution.state === 'errored') {
     throw globalWorld.execution.result;
   }
   if(currentWorld.execution.state === 'errored') {
     throw currentWorld.execution.result;
   }
   currentWorld.destroy();
+
+  await Promise.all(afterDestroyPromises);
```

Comment:
FYI - `afterEach` hooks are run inside-out; we can't wait for the child process to exit in the test's `afterEach` hook because we'd be waiting _before_ the current world is destroyed.

https://github.com/thefrontside/bigtest/pull/449/files#diff-b16dc92ebd232f17c34dba7e2a2f4f97R79

Reply:
In this case, maybe it would be best to just get a fresh port for each test so that we don't have to worry about collisions using something like [get-port](https://github.com/sindresorhus/get-port)?

Side-steps the issue without any special teardown since we know the port _is_ being released eventually.